packages/portfolio-contract/DESIGN-BETA.md

```diff
@@ -106,33 +106,92 @@ sequenceDiagram
 
   %% Notation: ->> for initial message, -->> for consequences
 
-  Note over EE: long-running process starts and monitors all portfolios
+  Note over YP: long-running process starts and monitors all portfolios
   Note over LCAin: USDC arriving in LCAin
-  LCAin -->> EE: EE observes USDC arrives
-  EE ->> LCAin: query bank balance
-  EE ->> portfolio: read portfolio allocations
-  EE ->> icaN: read balance
-  EE ->> AavePos: read balance
-  Note over EE: think and generate steps
-  EE ->> portfolio: send moves
+  LCAin -->> YP: YP observes USDC arrives
+  YP ->> LCAin: query bank balance
+  YP ->> portfolio: read portfolio allocations
+  YP ->> icaN: read balance
+  YP ->> aavePos: read balance
+  Note over YP: think and generate steps
+  YP ->> portfolio: send moves
   portfolio ->> LCAorch: CREATE (lazily)
-  %% Optionally, send to LCAin and have EE do the planning
-  critical CCTP out
-    LCAin ->> LCAorch: $5k
-    LCAorch ->> icaN: $5k
-    icaN ->> LCAorch: ack
-    LCAorch ->> icaN: depositForBurn
-    icaN ->> LCAorch: ack
-    icaN -->> acctArb: $5ku
-    acctArb -->> Res: event?
-    Res ->> portfolio: ack
-  option call `supply`
-    LCAorch ->> AX: supply $5k acctArb
-    AX -->> acctArb: supply $5k
-    acctArb ->> aavePos: $5k
-    aavePos -->> Res: Mint event
-    Res ->> portfolio: ack
+
+  Note over LCAorch, acctArb: CCTP Out
+  LCAin ->> LCAorch: $5k
+  LCAorch ->> icaN: $5k
+  icaN ->> LCAorch: ack
+  LCAorch ->> icaN: depositForBurn
+  icaN ->> LCAorch: ack
+  icaN -->> acctArb: $5ku
+  acctArb -->> Res: event?
+  Res ->> portfolio: ack
+
+  Note over LCAorch, aavePos: Supply to Aave
+  LCAorch ->> AX: supply $5k acctArb
+  AX -->> acctArb: supply $5k
+  acctArb ->> aavePos: $5k
+  aavePos -->> Res: Mint event
+  Res ->> portfolio: ack
+```
+
+5. manually-triggered rebalance
+```mermaid
+sequenceDiagram
+  autonumber
+
+  actor UI
+
+  box rgb(255,153,153) Agoric
+    participant portfolio
+    participant vstorage
+    participant LCAorch
+    participant LCAin
   end
+
+  box  rgb(255,165,0) Agoric off-chain
+    participant YP as Ymax Planner
+    participant Res as Resolver
+  end
+
+  box rgb(163,180,243) Noble
+      participant icaN as icaN*
+  end
+  box rgb(163,180,243) Axelar
+      participant AX as GMP
+  end
+  box rgb(163,180,243) Arbitrum
+      participant acctArb as acctArb*
+      participant aavePos as aavePos*
+  end
+  box rgb(163,180,243) Base
+      participant acctBase as acctBase*
+      participant compPos as compPos*
+  end
+
+  %% get plan
+  UI ->> portfolio: rebalance
+  portfolio ->> YP: YP observes portfolio allocation change
```

Comment:
allocation change? the rebalance use case doesn't (necessarily) involve an allocation change, does it?

Reply:
Yes, good catch, I'll have another PR to address this